pkg/storage/stores/indexshipper/downloads/index_set_test.go

```diff
@@ -88,10 +88,10 @@ func TestIndexSet_doConcurrentDownload(t *testing.T) {
 				require.Len(t, indexSet.index, tc)
 			}
 			verifyIndexForEach(t, indexesSetup, func(callbackFunc index.ForEachIndexCallback) error {
-				ctx, cancel := context.WithCancel(context.Background())
-				defer cancel()
+				doneChan := make(chan struct{})
+				defer close(doneChan)
 
-				return indexSet.ForEach(ctx, callbackFunc)
+				return indexSet.ForEach(context.Background(), doneChan, callbackFunc)
```

Comment:
Q: May be unrelated to this change. But what is the purpose of calling `ForEach` with already closed `donechan` here?